subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/OrderInsensitiveTaskFilePropertyCompareStrategy.java

```diff
@@ -38,9 +38,9 @@
 
 class OrderInsensitiveTaskFilePropertyCompareStrategy implements TaskFilePropertyCompareStrategy.Impl {
 
-    private static final Comparator<Entry<NormalizedFileSnapshot, IncrementalFileSnapshotWithAbsolutePath>> ENTRY_COMPARATOR = new Comparator<Entry<NormalizedFileSnapshot, IncrementalFileSnapshotWithAbsolutePath>>() {
+    private static final Comparator<Entry<String, IncrementalFileSnapshotWithAbsolutePath>> ENTRY_COMPARATOR = new Comparator<Entry<String, IncrementalFileSnapshotWithAbsolutePath>>() {
```

Comment:
I think this breaks things with `PathSensitivity.NONE`. I doesn't seem that we have a test for this. With path sensitivity `NONE`, all the relative paths are empty strings, so with this change here the strategy would compare if the number of elements are the same. So it would not detect a change in one of the hashes of the files anymore, wouldn't it?

Reply:
I pushed a test to `master` for this which breaks for your branch: https://github.com/gradle/gradle/commit/a614b4846247377c11b54fcdaabb2779adfef2c5

Reply:
Thank you @wolfs! I'll revert this and see what the performance impact is.

Reply:
It is about 1%. Maybe we can do something else. I'll request your review next time I have an idea in that area.